apollo-router/src/compute_job.rs

```diff
@@ -23,9 +24,47 @@ fn thread_pool_size() -> usize {
         .get()
 }
 
+/// Compute job queue is full
+#[derive(thiserror::Error, Debug, displaydoc::Display, Clone)]
+pub(crate) struct ComputeBackPressureError;
+
+#[derive(Debug)]
+pub(crate) enum MaybeBackPressureError<E> {
+    /// Doing the same request again later would result in the same error (e.g. invalid query).
+    ///
+    /// This error can be cached.
+    PermanentError(E),
+
+    /// Doing the same request again later might work.
+    ///
+    /// This error must not be cached.
+    TemporaryError(ComputeBackPressureError),
+}
```

Comment:
We should definitely have a metric for both of these. Even just for us to know how many of the requests get turned away because the job queue is full so we can validate this work.

And speaking of, should we also have a metric to count the number of jobs in the queue? Also for our own "selfish" reasons to see if our assumptions are correct.

Reply:
The `PermanentError` case is generic and used in a lot of existing places, so there isn’t a single thing to count. I’m adding a counter metric for when `compute_job::execute()` returns `ComputeBackPressureError`.

We already have a `apollo.router.compute_jobs.queued` gauge metric.

Reply:
thank you!!